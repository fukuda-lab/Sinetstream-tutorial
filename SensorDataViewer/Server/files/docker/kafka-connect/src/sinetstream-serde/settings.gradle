rootProject.name = 'sinetstream-kafka-serde'
