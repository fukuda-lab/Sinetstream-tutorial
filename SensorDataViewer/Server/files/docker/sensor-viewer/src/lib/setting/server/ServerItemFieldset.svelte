<script lang="ts" context="module">
  export const initTargets = [
    "player.from",
    "android",
    "android.publisher",
    "image",
    "image.topic",
    "perftool",
    "perftool.name",
  ];
</script>

<script lang="ts">
  import type { ViewerSettingV1 } from "src/lib/settings";
  import HorizontalLayoutIcon from "../layout/HorizontalLayoutIcon.svelte";
  import Vertical2LayoutIcon from "../layout/Vertical2LayoutIcon.svelte";
  import VerticalHorizontalLayoutIcon from "../layout/VerticalHorizontalLayoutIcon.svelte";
  import VerticalLayoutIcon from "../layout/VerticalLayoutIcon.svelte";
  import ItemCheckbox from "./ItemCheckbox.svelte";
  import SubItemCheckbox from "./SubItemCheckbox.svelte";
  import SubItemLabel from "./SubItemLabel.svelte";

  export let initValues: string[] = initTargets;
  export let setting: ViewerSettingV1 | null;
</script>

<fieldset aria-label="target list of settings">
  {#if setting}
    {@const {
      from: playerFrom,
      to: playerTo,
      ...playerParams
    } = setting?.player ?? {}}
    {@const { publisher, ...androidParams } = setting?.android ?? {}}
    {@const { topic, ...imageParams } = setting?.image ?? {}}
    {@const { name: perftoolName, ...perftoolParams } = setting?.perftool ?? {}}
    {@const { layout, chart: chartParams } = setting ?? {}}

    <ItemCheckbox value={"player"}>
      <span slot="label">Playback</span>
      <SubItemCheckbox name={"from"}>
        From / To:
        {JSON.stringify(playerFrom) ?? ""} ---
        {JSON.stringify(playerTo) ?? ""}
      </SubItemCheckbox>

      <SubItemLabel value={playerParams?.tick}>Refresh Interval</SubItemLabel>
      <SubItemLabel value={playerParams?.speed}>Resolution</SubItemLabel>
      <SubItemLabel value={playerParams?.repeat}>Repeat</SubItemLabel>
    </ItemCheckbox>

    <ItemCheckbox value={"android"}>
      <span slot="label">Android Sensors</span>
      <SubItemCheckbox value={publisher} name={"publisher"}>
        Publisher
      </SubItemCheckbox>
      <SubItemLabel value={androidParams?.map}>Map</SubItemLabel>
      <SubItemLabel value={androidParams?.lte}>LTE Chart</SubItemLabel>
      <SubItemLabel value={androidParams?.sensors}>Sensors</SubItemLabel>
    </ItemCheckbox>

    <ItemCheckbox value={"image"}>
      <span slot="label">Raspberry Pi Camera</span>
      <SubItemCheckbox value={topic} name={"topic"}>Topic Name</SubItemCheckbox>
      <SubItemLabel value={imageParams?.visible}>Visible</SubItemLabel>
      <SubItemLabel value={imageParams?.update}>Update Method</SubItemLabel>
    </ItemCheckbox>

    <ItemCheckbox value={"perftool"}>
      <span slot="label">Perftool</span>
      <SubItemCheckbox value={perftoolName} name={"name"}>
        Perftool Name
      </SubItemCheckbox>
      <SubItemLabel value={perftoolParams?.visible}>Visible</SubItemLabel>
    </ItemCheckbox>

    <ItemCheckbox value={"layout"}>
      <span slot="label">Layout</span>
      <SubItemLabel>
        {#if layout?.template === "horizontal"}
          <HorizontalLayoutIcon />
        {:else if layout?.template === "vertical-horizontal"}
          <VerticalHorizontalLayoutIcon />
        {:else if layout?.template === "vertical"}
          <VerticalLayoutIcon />
        {:else if layout?.template === "vertical-2"}
          <Vertical2LayoutIcon />
        {/if}
      </SubItemLabel>
    </ItemCheckbox>

    <ItemCheckbox value={"chart"}>
      <span slot="label">Chart</span>
      <SubItemLabel value={chartParams?.line?.enabled}>
        Line Visible
      </SubItemLabel>
      <SubItemLabel value={chartParams?.line?.size}>Line Width</SubItemLabel>
      <SubItemLabel value={chartParams?.point?.enabled}>
        Point Visible
      </SubItemLabel>
      <SubItemLabel value={chartParams?.point?.size}>Point Size</SubItemLabel>
      <SubItemLabel value={chartParams?.minSize?.width}>
        Minimum Width
      </SubItemLabel>
      <SubItemLabel value={chartParams?.minSize?.height}>
        Minimum Height
      </SubItemLabel>
    </ItemCheckbox>
  {:else}
    {#each initValues as value}
      <input type="checkbox" name="targets" {value} />
    {/each}
  {/if}
</fieldset>

<style>
  fieldset {
    border-color: transparent;
  }
</style>
