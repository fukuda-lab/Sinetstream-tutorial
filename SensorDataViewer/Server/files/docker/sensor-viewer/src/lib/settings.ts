import { format, parseISO, sub } from 'date-fns';
import { persisted } from 'svelte-local-storage-store';
import {
  derived, writable, type Readable, type Writable,
} from 'svelte/store';
import * as zod from 'zod';

const paneVisibleSchema = zod.object({
  map: zod.boolean().optional(),
  image: zod.boolean().optional(),
  perftool: zod.boolean().optional(),
  lte: zod.boolean().optional(),
  sensors: zod.array(zod.string()).optional(),
});
const playerSchema = zod.object({
  from: zod.union([zod.preprocess(
    (arg) => (typeof arg === 'string' ? parseISO(arg) : arg),
    zod.date(),
  ), zod.string().regex(/^\d{4}-\d{2}-\d{2} \d{2}:\d{2}$/)]),
  to: zod.union([zod.preprocess(
    (arg) => (typeof arg === 'string' ? parseISO(arg) : arg),
    zod.date(),
  ), zod.string().regex(/^\d{4}-\d{2}-\d{2} \d{2}:\d{2}$/)]),
  tick: zod.number().gte(100),
  speed: zod.number().gte(1).lte(100),
  repeat: zod.boolean(),
});
const chartV0Schema = zod.object({
  minSize: zod.object({
    width: zod.number().gte(200).optional(),
    height: zod.number().gte(150).optional(),
  }).optional(),
  line: zod.object({
    enabled: zod.boolean().optional(),
    size: zod.number().gt(0).optional(),
  }).optional(),
  point: zod.object({
    enabled: zod.boolean().optional(),
    size: zod.number().gt(0).optional(),
  }).optional(),
});
const layoutTemplateSchema = zod.enum(['horizontal', 'vertical-horizontal', 'vertical', 'vertical-2']);
const viewerSettingSchemaV0 = zod.object({
  name: zod.string(),
  publisher: zod.string().optional(),
  imageTopic: zod.string().optional(),
  perftoolName: zod.string().optional(),
  panes: paneVisibleSchema.optional(),
  imageUpdate: zod.enum(['push', 'polling']).optional(),
  player: playerSchema.optional(),
  layout: zod.object({ template: layoutTemplateSchema }).optional(),
  chart: chartV0Schema.optional(),
});
const androidSchema = zod.object({
  publisher: zod.string(),
  map: zod.boolean(),
  lte: zod.boolean(),
  sensors: zod.array(zod.string()),
});
const picameraSchema = zod.object({
  topic: zod.string(),
  update: zod.enum(['push', 'polling']),
  visible: zod.boolean(),
});
const layoutSchema = zod.object({
  template: layoutTemplateSchema,
});
const perftoolSchema = zod.object({
  name: zod.string(),
  visible: zod.boolean(),
});
const chartV1Schema = zod.object({
  minSize: zod.object({
    width: zod.number().gte(200),
    height: zod.number().gte(150),
  }),
  line: zod.object({
    enabled: zod.boolean(),
    size: zod.number().gt(0),
  }),
  point: zod.object({
    enabled: zod.boolean(),
    size: zod.number().gt(0),
  }),
});
const viewerSettingSchemaV1 = zod.object({
  name: zod.string(),
  version: zod.literal('1.0'),
  android: androidSchema,
  chart: chartV1Schema,
  image: picameraSchema,
  layout: layoutSchema,
  perftool: perftoolSchema,
  player: playerSchema,
});

type ViewerSettingV0 = zod.infer<typeof viewerSettingSchemaV0>;
type ViewerSettingV1 = zod.infer<typeof viewerSettingSchemaV1>;
type ViewerSetting = ViewerSettingV0 | ViewerSettingV1;
type PaneVisible = zod.infer<typeof paneVisibleSchema>;
type SensorViewerMode = 'viewer' | 'player';
type LayoutType = 'horizontal' | 'vertical-horizontal' | 'vertical' | 'vertical-2';
type ImageUpdateMethod = 'push' | 'polling';

function isViewerSettingV0(value: unknown): value is ViewerSettingV0 {
  return viewerSettingSchemaV0.safeParse(value).success;
}

function isViewerSettingV1(value: unknown): value is ViewerSettingV1 {
  return viewerSettingSchemaV1.safeParse(value).success;
}

const mode: Writable<SensorViewerMode> = persisted('mode', 'viewer');
const settings: Writable<ViewerSetting[]> = persisted('settings', []);
const settingsIndex = writable(0);

function defaultDateTime(minutes = 0): string {
  return format(sub(new Date(), { minutes }), 'yyyy-MM-dd HH:mm');
}

function toV1(v0: ViewerSettingV0): ViewerSettingV1 {
  return {
    name: v0.name,
    version: '1.0',
    android: {
      publisher: v0.publisher ?? '',
      map: v0.panes?.map ?? true,
      lte: v0.panes?.lte ?? true,
      sensors: v0.panes?.sensors ?? [],
    },
    chart: {
      minSize: {
        width: v0.chart?.minSize?.width ?? 310,
        height: v0.chart?.minSize?.height ?? 250,
      },
      line: {
        enabled: v0.chart?.line?.enabled ?? true,
        size: v0.chart?.line?.size ?? 1,
      },
      point: {
        enabled: v0.chart?.point?.enabled ?? true,
        size: v0.chart?.point?.size ?? 2,
      },
    },
    image: {
      topic: v0.imageTopic ?? '',
      update: v0.imageUpdate ?? 'push',
      visible: v0.panes?.image ?? true,
    },
    layout: {
      template: v0.layout?.template ?? 'horizontal',
    },
    perftool: {
      name: v0.perftoolName ?? '',
      visible: v0.panes?.perftool ?? true,
    },
    player: {
      from: v0.player?.from ?? defaultDateTime(15),
      to: v0.player?.to ?? defaultDateTime(),
      tick: v0.player?.tick ?? 1200,
      speed: v0.player?.tick ?? 10,
      repeat: v0.player?.repeat ?? false,
    },
  };
}

function defaultSetting(settingName: string): ViewerSettingV1 {
  return toV1({ name: settingName });
}

const currentSetting: Readable<ViewerSettingV1> = derived(
  [settings, settingsIndex],
  ([$settings, $settingsIndex], set) => {
    if ($settingsIndex < $settings.length && $settingsIndex >= 0) {
      const current = $settings[$settingsIndex];
      if (isViewerSettingV1(current)) {
        set(current);
      } else if (isViewerSettingV0(current)) {
        set(toV1(current));
      }
    } else {
      set(defaultSetting(''));
    }
  },
);

const name: Readable<string> = derived([currentSetting], ([$currentSetting], set) => {
  set($currentSetting.name);
});

const publisher: Readable<string> = derived([currentSetting], ([$currentSetting], set) => {
  set($currentSetting.android.publisher);
});

const perftoolName: Readable<string> = derived([currentSetting], ([$currentSetting], set) => {
  set($currentSetting.perftool.name);
});

const imageTopic: Readable<string> = derived([currentSetting], ([$currentSetting], set) => {
  set($currentSetting.image.topic);
});

const pane: Readable<PaneVisible> = derived([currentSetting], ([$currentSetting], set) => {
  const current = $currentSetting;
  set({
    lte: current.android.lte,
    map: current.android.map,
    sensors: current.android.sensors,
    image: current.image.visible,
    perftool: current.perftool.visible,
  });
});

const charts: Readable<string[]> = derived([currentSetting], ([$currentSetting], set) => {
  const current = $currentSetting;
  const { sensors } = current.android;
  const result: string[] = [...sensors];
  if (current.android.lte) {
    result.unshift('lte');
  }
  if (current.perftool.visible) {
    result.unshift('perftool');
  }
  set(result);
});

const imageUpdateMethod: Readable<ImageUpdateMethod> = derived(
  [currentSetting],
  ([$currentSetting], set) => {
    set($currentSetting.image.update);
  },
);

export {
  settings,
  settingsIndex,
  mode,
  name,
  publisher,
  perftoolName,
  imageTopic,
  pane,
  charts,
  imageUpdateMethod,
  currentSetting,
  isViewerSettingV0,
  isViewerSettingV1,
  viewerSettingSchemaV1,
};
export type {
  SensorViewerMode,
  ImageUpdateMethod,
  ViewerSetting,
  ViewerSettingV0,
  ViewerSettingV1,
  PaneVisible,
  LayoutType,
};
