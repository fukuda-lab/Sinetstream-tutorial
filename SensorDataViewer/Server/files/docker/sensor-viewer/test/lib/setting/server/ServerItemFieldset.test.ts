import { cleanup, render } from '@testing-library/svelte';
import { writable } from 'svelte/store';
import {
  afterEach, describe, expect, it,
} from 'vitest';
import ServerItemFieldset from '../../../../src/lib/setting/server/ServerItemFieldset.svelte';
import { ImageUpdateMethod, LayoutType, ViewerSettingV1 } from '../../../../src/lib/settings';
import ItemContextTest from './ItemContextTest.svelte';

describe('ServerItemFieldset.svelt', () => {
  afterEach(() => cleanup());

  it('empty', () => {
    const targets = writable([]);
    const setting = {};
    const { container, getByRole } = render(ItemContextTest, {
      props: {
        Component: ServerItemFieldset,
        targets,
        props: { setting },
        value: 'xxx',
        emptySlot: true,
      },
    });
    expect(getByRole('group')).toBeTruthy();
    expect(container.outerHTML).toMatchSnapshot();
  });

  it('setting', () => {
    const targets = writable([]);
    const player = {
      from: '2023-02-01 00:00',
      to: '2023-02-03 14:30',
      tick: 1200,
      speed: 10,
      repeat: false,
    };
    const android = {
      publisher: 'pub-1',
      map: true,
      lte: true,
      sensors: [],
    };
    const image = {
      topic: 'topic-1',
      update: 'push' as ImageUpdateMethod,
      visible: true,
    };
    const perftool = {
      name: 'perf-1',
      visible: true,
    };
    const layout = {
      template: 'horizontal' as LayoutType,
    };
    const chart = {
      minSize: {
        width: 300,
        height: 200,
      },
      point: {
        size: 3,
        enabled: true,
      },
      line: {
        size: 3,
        enabled: true,
      },
    };
    const setting: ViewerSettingV1 = {
      name: 'setting-1',
      version: '1.0',
      android,
      image,
      perftool,
      player,
      layout,
      chart,
    };
    const { container, getByRole } = render(ItemContextTest, {
      props: {
        Component: ServerItemFieldset,
        targets,
        props: { setting },
        value: 'xxx',
        emptySlot: true,
      },
    });
    expect(getByRole('group')).toBeTruthy();
    expect(container.outerHTML).toMatchSnapshot();
  });

  it('initValues', () => {
    const targets = writable([]);
    const setting = null;
    const { container, getByRole } = render(ItemContextTest, {
      props: {
        Component: ServerItemFieldset,
        targets,
        props: { setting },
        value: 'xxx',
        emptySlot: true,
      },
    });
    expect(getByRole('group')).toBeTruthy();
    expect(container.outerHTML).toMatchSnapshot();
  });
});
