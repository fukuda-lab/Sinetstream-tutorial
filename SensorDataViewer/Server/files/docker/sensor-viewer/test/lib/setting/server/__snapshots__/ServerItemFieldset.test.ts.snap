// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerItemFieldset.svelt > empty 1`] = `
"<body><div><fieldset aria-label=\\"target list of settings\\" class=\\"s-1zY7IsgIXyl5\\"><div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"player\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Playback</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"player.from\\" aria-label=\\"player.from\\"> <span></span></label> <div>From / To:
         ---
        </div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Refresh Interval</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Resolution</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Repeat</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"android\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Android Sensors</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"android.publisher\\" aria-label=\\"android.publisher\\"> <span></span></label> <div>Publisher</div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Map</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>LTE Chart</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Sensors</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"image\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Raspberry Pi Camera</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"image.topic\\" aria-label=\\"image.topic\\"> <span></span></label> <div>Topic Name</div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Visible</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Update Method</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"perftool\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Perftool</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"perftool.name\\" aria-label=\\"perftool.name\\"> <span></span></label> <div>Perftool Name</div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Visible</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"layout\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Layout</span></div></div> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div></div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"chart\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Chart</span></div></div> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Line Visible</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Line Width</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Point Visible</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Point Size</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Minimum Width</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Minimum Height</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--></fieldset><!--<ServerItemFieldset>--><!--<ItemContextTest>--></div></body>"
`;

exports[`ServerItemFieldset.svelt > initValues 1`] = `"<body><div><fieldset aria-label=\\"target list of settings\\" class=\\"s-1zY7IsgIXyl5\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"player.from\\" class=\\"s-1zY7IsgIXyl5\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"android\\" class=\\"s-1zY7IsgIXyl5\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"android.publisher\\" class=\\"s-1zY7IsgIXyl5\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"image\\" class=\\"s-1zY7IsgIXyl5\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"image.topic\\" class=\\"s-1zY7IsgIXyl5\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"perftool\\" class=\\"s-1zY7IsgIXyl5\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"perftool.name\\" class=\\"s-1zY7IsgIXyl5\\"></fieldset><!--<ServerItemFieldset>--><!--<ItemContextTest>--></div></body>"`;

exports[`ServerItemFieldset.svelt > setting 1`] = `
"<body><div><fieldset aria-label=\\"target list of settings\\" class=\\"s-1zY7IsgIXyl5\\"><div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"player\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Playback</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"player.from\\" aria-label=\\"player.from\\"> <span></span></label> <div>From / To:
        \\"2023-02-01 00:00\\" ---
        \\"2023-02-03 14:30\\"</div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Refresh Interval: 1200</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Resolution: 10</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Repeat: false</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"android\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Android Sensors</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"android.publisher\\" aria-label=\\"android.publisher\\"> <span></span></label> <div>Publisher: \\"pub-1\\"</div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Map: true</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>LTE Chart: true</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Sensors: []</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"image\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Raspberry Pi Camera</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"image.topic\\" aria-label=\\"image.topic\\"> <span></span></label> <div>Topic Name: \\"topic-1\\"</div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Visible: true</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Update Method: \\"push\\"</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"perftool\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Perftool</span></div></div> <div class=\\"row\\"><div></div> <label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"perftool.name\\" aria-label=\\"perftool.name\\"> <span></span></label> <div>Perftool Name: \\"perf-1\\"</div></div><!--<SubItemCheckbox>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Visible: true</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"layout\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Layout</span></div></div> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div><span><table border=\\"1\\"><tr><td><i>map</i></td> <td><i>image</i></td></tr> <tr><td colspan=\\"2\\" style=\\"text-align: center;\\"><i>show_chart</i></td></tr></table></span><!--<HorizontalLayoutIcon>--></div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--> <div class=\\"row\\"><label class=\\"checkbox\\"><input type=\\"checkbox\\" name=\\"targets\\" value=\\"chart\\"> <span></span></label> <div><span slot=\\"label\\" class=\\"s-1zY7IsgIXyl5\\">Chart</span></div></div> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Line Visible: true</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Line Width: 3</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Point Visible: true</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Point Size: 3</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Minimum Width: 300</div></div><!--<SubItemLabel>--> <div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>Minimum Height: 200</div></div><!--<SubItemLabel>--><!--<ItemCheckbox>--></fieldset><!--<ServerItemFieldset>--><!--<ItemContextTest>--></div></body>"
`;
